homeassistant/components/knx/__init__.py

```diff
@@ -235,25 +256,56 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     async def reload_service_handler(service_call: ServiceCall) -> None:
         """Remove all KNX components and load new ones from config."""
+        await hass.config_entries.async_reload(entry.entry_id)
```

Comment:
Since you're not reloading configuration.yaml, you indeed no longer need this.

Reply:
Does the config entry reload not reload the YAML?

Reply:
Hm. But we want to reload the YAML, I thought the config entry reload will also reload the YAML, so that's not true?

Reply:
It does not reload the YAML. I adjusted the reload of the config entry to also reload the YAML. Otherwise, what exactly is the reason a user would ever want to reload.